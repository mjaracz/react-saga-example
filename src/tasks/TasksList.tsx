import React, { FC } from 'react'
import { useTasksList } from './hooks/useTasksList'
import { Task } from './Task'
import { CircularProgress, List, ListItemIcon, ListSubheader } from '@material-ui/core'
import { AddOutlined } from '@material-ui/icons'
import { AddTask } from './AddTask'
import './styles/tasks-list.css'

export const TasksList: FC = () => {
  const { tasksStore, handleAddTaskIcon, isOpenModal, setIsOpenModal } = useTasksList()

  if (tasksStore.isLoading) {
    return (
      <CircularProgress className="tasks--circular" size={120}/>
    )
  }

  return (
    <List
      component="main"
      className="tasks"
      aria-labelledby="subheader"
      subheader={
        <ListSubheader className="tasks__subheader" component="div" id="subheader">
          Default To-Do list, add your tasks
        </ListSubheader>
      }
    >
      {
        tasksStore.tasks.map((task, index) =>
          <Task key={index} id={index} title={task.title} description={task?.description}/>
        )
      }
      <ListItemIcon className="tasks__addTaskIcon" onClick={handleAddTaskIcon}>
        <AddOutlined/>
      </ListItemIcon>
      <AddTask isOpen={isOpenModal} handleAddTaskIcon={handleAddTaskIcon} setIsOpenModal={setIsOpenModal} />
    </List>
  )
}
