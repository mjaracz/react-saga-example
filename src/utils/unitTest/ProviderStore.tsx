import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

export const ProviderStore: FC = ({ children }) => {
  return (
    <Provider store={store}>
      { children }
    </Provider>
  )
}
