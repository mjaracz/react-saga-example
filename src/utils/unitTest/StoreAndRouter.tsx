import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../redux/store'

export const StoreAndRouter: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        { children }
      </BrowserRouter>
    </Provider>
  )
}
