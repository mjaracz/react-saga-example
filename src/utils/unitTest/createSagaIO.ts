import { stdChannel } from 'redux-saga'
import { EventEmitter } from 'events'


export const createSagaIO = (emitter: EventEmitter) => {
  const channel = stdChannel()
  emitter.on('action', channel.put)

  return {
    channel,
    dispatch: action => emitter.emit('action', action),
    getState: () => ({})
  }
}
