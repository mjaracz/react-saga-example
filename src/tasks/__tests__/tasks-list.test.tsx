import React from 'react'
import { render } from '@testing-library/react'
import { TasksList } from '../TasksList'
import { StoreAndRouter } from '../../utils/unitTest/StoreAndRouter'

describe('<TasksList />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <StoreAndRouter>
        <TasksList/>
      </StoreAndRouter>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
