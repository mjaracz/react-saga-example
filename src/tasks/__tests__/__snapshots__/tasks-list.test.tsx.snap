// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiCircularProgress-root tasks--circular MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 120px; height: 120px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</body>
`;
